lib/src/configuration.cc

```diff
@@ -156,6 +161,11 @@ HW::ParseResult Configuration::parseOptions(int argc, char *argv[])
     // This is primarily for testing.
     master_uris_.clear();
 
+    // remember the path to the pxp-agent executable used to start
+    // this process, it is supposed to be used to start executables
+    // which are installed alongside the pxp-agent executable
+    exec_prefix_ = fs::absolute(fs::path(argv[0]).parent_path());
```

Comment:
Unfortunately if the `argv[0]` doesn't contain any path (only the filename of the executable) then the executable was most probably found in one of the directories specified in the `PATH` environment variable.
In this scenario the
```c++
fs::path(argv[0]).parent_path()
```
expression returns an empty fs::path and calling `fs::absolute()` on such an empty fs::path produces a fs::path representation of the current working directory which is most likely not the directory where the pxp-agent excutable is installed.

Therefore we should drop the call to `fs::absolute()` and instead remember the `exec_prefix_` path as returned by the aforementioned expression:
```c++
fs::path(argv[0]).parent_path()
```
Since we don't change the current working directory in pxp-agent, this should be safe to use to execute executables colocated with the pxp-agent executable.

Alternatively we could search for the pxp-agent executable in the directories in the `PATH` manually.